### ISSUE TITLE
rubocop's rubocop config is missing a cop

### ISSUE THREAD

**Issue by logicminds** [bbatsov/rubocop#3431]
My PR #3365 was checked by the cop, but there seems to be missing a config for 
methods that require a newline above/below.

ie.

```
HTTP_METHODS = [:get, :post, :put, :patch, :delete, :head].freeze
def on_send(node)
   do something
end

```

**Comment by bbatsov** [comment]
Yep, this looks like a bug.

**Comment by tejasbubane** [comment]
The `Style/EmptyLineBetweenDefs` [does not handle this case](https://github.com/bbatsov/rubocop/blob/master/spec/rubocop/cop/style/empty_line_between_defs_spec.rb#L158-L166). Or is there any other cop that I am missing on?

**Comment by mikegee** [comment]
This issue seems like either a misunderstanding of the expected behavior of `Style/EmptyLineBetweenDefs` or a request to change it, not a bug. Is that right?

**Comment by Drenmi** [comment]
@mikegee: Yes. Sounds like a request for `Layout/EmptyLinesAroundMethodDefinition`. Such a cop would probably also need to be able to account for modifiers, which are commonly placed adjacent to the definition, e.g.:

```ruby
def foo
  ...
end
private :foo
```
